starter_code/src/movies.js

```diff
@@ -2,12 +2,55 @@
 // Turn duration of the movies from hours to minutes 
 
 
+
+
 // Get the average of all rates with 2 decimals 
 
 
+
+
+function ratesAverage(film) {
+    var average = film.reduce(function (sum, item) {
+      return sum + parseFloat(item.rate);
+    }, 0) / film.length;
+
+    return Math.round(average*100)/100
+  }
+  
+  ratesAverage(movies);
+  console.log(ratesAverage(movies));
+  
 // Get the average of Drama Movies
 
 
+//var drama = movies.filter(function(movies) {
+  //  return movies.genre.includes("Drama");
+//});
+
+
+
+function dramaMoviesRate(movie) {
+    var drama = movie.filter(function(movie) {
+        return movie.genre.includes("Drama");
+    }); 
+
+    if (drama.length === 0) {
+        return;
+    }
+
+    var average = drama.reduce(function (sum, item) {
```

Comment:
Don't you feel like you are repeating yourselves? After filtering the array, you can reuse the function that you write before (ratesAverage) to calculate the average.